lib/table/normalize.js

```diff
@@ -0,0 +1,198 @@
+ *
+ * @this PDFTable
+ * @memberOf PDFTable
+ * @private
+ */
+export function normalizeTable() {
+  // Normalize config
+  let index = this.document._tableIndex++;
+  this._id = new String(this.opts.id ?? `table-${index}`);
+  this._position = {
+    x: this.document.sizeToPoint(this.opts.position?.x, this.document.x),
+    y: this.document.sizeToPoint(this.opts.position?.y, this.document.y),
+  };
+  this._maxWidth = this.document.sizeToPoint(
+    this.opts.maxWidth,
+    this.document.page.width -
+      this.document.page.margins.right -
+      this._position.x,
+  );
+
+  const { defaultStyle, defaultColStyle, defaultRowStyle } =
+    normalizedDefaultStyle(this.opts.defaultStyle);
+  this._defaultStyle = defaultStyle;
+
+  let colStyle;
+  if (this.opts.columnStyles) {
+    if (Array.isArray(this.opts.columnStyles)) {
+      colStyle = (i) => this.opts.columnStyles[i];
+    } else if (typeof this.opts.columnStyles === 'function') {
+      colStyle = this.opts.columnStyles;
+    } else if (typeof this.opts.columnStyles === 'object') {
+      colStyle = () => this.opts.columnStyles;
+    }
+  }
+  if (!colStyle) colStyle = () => ({});
+  this._colStyle = normalizedColumnStyle.bind(this, defaultColStyle, colStyle);
+
+  let rowStyle;
+  if (this.opts.rowStyles) {
+    if (Array.isArray(this.opts.rowStyles)) {
+      rowStyle = (i) => this.opts.rowStyles[i];
+    } else if (typeof this.opts.rowStyles === 'function') {
+      rowStyle = this.opts.rowStyles;
+    } else if (typeof this.opts.rowStyles === 'object') {
+      rowStyle = () => this.opts.rowStyles;
+    }
+  }
+  if (!rowStyle) rowStyle = () => ({});
+  this._rowStyle = normalizedRowStyle.bind(this, defaultRowStyle, rowStyle);
+}
+
+/**
+ * Convert text into a string
+ * - null and undefined are preserved (as they will be ignored)
+ * - everything else is run through `String()`
+ *
+ * @param {*} text
+ * @returns {string}
+ * @private
+ */
+export function normalizeText(text) {
+  // Parse out text
+  if (text !== null && text !== undefined) text = String(text);
+  return text;
+}
+
+/**
+ * Normalize a cell config
+ *
+ * @this PDFTable
+ * @memberOf PDFTable
+ * @param {TableCellStyle} cell - The cell to mutate
+ * @param {number} rowIndex - The cells row
+ * @param {number} colIndex - The cells column
+ * @returns {NormalizedTableCellStyle}
+ * @private
+ */
+export function normalizeCell(cell, rowIndex, colIndex) {
+  const colStyle = this._colStyle(colIndex);
+  let rowStyle = this._rowStyle(rowIndex);
+
+  const font = deepMerge({}, colStyle.font, rowStyle.font, cell.font);
+
+  // Initialize cell context
+  const rollbackFont = this.document._fontSource;
+  const rollbackFontSize = this.document._fontSize;
+  const rollbackFontFamily = this.document._fontFamily;
+  if (font.src) this.document.font(font.src, font.family);
+  if (font.size) this.document.fontSize(font.size);
+
+  // Refetch rowStyle to reflect font changes
+  rowStyle = this._rowStyle(rowIndex);
```

Comment:
Why need to be refetched here?

Other way to ask: how setting document font affects _rowStyle result?

Reply:
since the dynamic sizing stuff it is possible to be based on the fontSize for some fields. (e.g. padding) where a user may want the padding of a cell to be `1em` (1 fontSize) but this padding should be of the cells font not the document (this is the same behaviour as html tables)

Let me know if you need more clarification

Reply:
Ok. 

When no font is defined, the refetch is not needed. Do you think is worth to refetch only when font is defined in one of the styles?

Reply:
i've now made it so that the _rowStyle and _colStyle is memoized so the double call doesnt really affect anything

Reply:
I've decided to apply the refetch skipping logic